src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -129,6 +146,30 @@ public void setDirectoryScanner( DirectoryScanner directoryScanner )
         this.directoryScanner = directoryScanner;
     }
 
+    /**
+     * Get the existing DirectoryScanner used by this helper,
+     * or create new a DirectoryScanner if none is yet set.
+     * The DirectoryScanner is used for detecting changes in a directory
+     */
+    public DirectoryScanner getGeneratedSourcesDirectoryScanner()
+    {
+        if ( generatedSourcesDirectoryScanner == null )
+        {
+            generatedSourcesDirectoryScanner = new DirectoryScanner();
+        }
+
+        return generatedSourcesDirectoryScanner;
+    }
+
+    /**
+     * Set the DirectoryScanner which shall get used by this build helper.
```

Comment:
which shall get used by this build helper --> that scans the target/generated sources directory

Reply:
Oops yeah just copy/pasted the javadocs. I will update them.